### PR TITLE
MAINT: more prep for deprecation of old-style data_store classes

### PR DESCRIPTION
**Pull request by GavinHuttley**

### CODE REVIEW

File: src/cogent3/app/data_store_new.py [hunk 1]
```diff
@@ -502,15 +506,23 @@ def write_log(self, *, unique_id: str, data: str) -> None:
 def get_data_source(data) -> str:
```
Comment (Nick-Foto) [inline src/cogent3/app/data_store_new.py:506]: When I look at this as a user, I get confused about naming. I think other names like `get_source_path` or `get_data_path` or a different name will be more informative.

File: src/cogent3/app/data_store_new.py [hunk 2]
```diff
@@ -502,15 +506,23 @@ def write_log(self, *, unique_id: str, data: str) -> None:
...
     source = getattr(data, "source", None)
     if source is None:
```
Comment (Nick-Foto) [inline src/cogent3/app/data_store_new.py:508]: I think our new datastores doesn't have `source` attribute. Why we use this attribute name here?

Comment (GavinHuttley) [reply src/cogent3/app/data_store_new.py:508]: this is for data objects, it's what the unique ID is derived from

File: src/cogent3/app/data_store_new.py (hunk 1 above)
Comment (GavinHuttley) [reply src/cogent3/app/data_store_new.py:506]: Fair point. The biggest issue is not enough docstrings.

**Pull request merged**